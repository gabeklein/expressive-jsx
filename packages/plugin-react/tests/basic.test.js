const test = require("./_adapter");

test("component", `
  () => do {
    <div>Hello World</div>
  }
`);

test("multiple", `
  ({ hello }) => do {
    <this>
      <div>Foo</div>
      <div>Bar</div>
    </this>
  }
`);

test("switch", `
  ({ hello }) => do {
    if(hello)
      <div>Hello World</div>
    else
      <div>Goodbye World</div>
  }
`);