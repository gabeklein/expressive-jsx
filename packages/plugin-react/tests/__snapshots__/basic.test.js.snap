// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component 1`] = `"() => create(\\"div\\", {}, \\"Hello World\\");"`;

exports[`component 2`] = `"() => <div>Hello World</div>;"`;

exports[`multiple 1`] = `
"({
  hello
}) => create(Fragment, {}, create(\\"div\\", {}, \\"Foo\\"), create(\\"div\\", {}, \\"Bar\\"));"
`;

exports[`multiple 2`] = `
"({
  hello
}) => <Fragment><div>Foo</div><div>Bar</div></Fragment>;"
`;

exports[`switch 1`] = `
"({
  hello
}) => create(\\"div\\", {
  \\"className\\": join(\\"do_erd\\", false)
}, hello ? create(\\"div\\", {}, \\"Hello World\\") : create(\\"div\\", {}, \\"Goodbye World\\"));"
`;

exports[`switch 2`] = `
"({
  hello
}) => <div className={join(\\"do_erd\\", false)}>{hello ? <div>Hello World</div> : <div>Goodbye World</div>}</div>;"
`;
